docarray/typing/tensor/abstract_tensor.py

```diff
@@ -60,26 +72,43 @@ class AbstractTensor(Generic[ShapeT], AbstractType, ABC):
     _PROTO_FIELD_NAME: str
 
     @classmethod
-    @abc.abstractmethod
-    def __docarray_validate_shape__(cls, t: T, shape: Tuple[int]) -> T:
-        """Every tensor has to implement this method in order to
-        enable syntax of the form AnyTensor[shape].
-
-        It is called when a tensor is assigned to a field of this type.
-        i.e. when a tensor is passed to a Document field of type AnyTensor[shape].
-
-        The intended behaviour is as follows:
-        - If the shape of `t` is equal to `shape`, return `t`.
-        - If the shape of `t` is not equal to `shape`,
-            but can be reshaped to `shape`, return `t` reshaped to `shape`.
-        - If the shape of `t` is not equal to `shape`
-            and cannot be reshaped to `shape`, raise a ValueError.
-
-        :param t: The tensor to validate.
-        :param shape: The shape to validate against.
-        :return: The validated tensor.
-        """
-        ...
+    def __docarray_validate_shape__(
+        cls, t: T, shape: Tuple[Union[int, str]]
+    ) -> T:  # type: ignore
```

Comment:
why is it type ignore her e?

Reply:
Ah its an artifact from copying the signature in TorchTensor. Seems to work fine without it. Removed